src/joystick/SDL_gamepad.c

```diff
@@ -1105,29 +1105,31 @@ const char *SDL_GetGamepadStringForAxis(SDL_GamepadAxis axis)
 }
 
 static const char *map_StringForGamepadButton[] = {
-    "a",
-    "b",
-    "x",
-    "y",
-    "back",
-    "guide",
-    "start",
-    "leftstick",
-    "rightstick",
-    "leftshoulder",
-    "rightshoulder",
-    "dpup",
-    "dpdown",
-    "dpleft",
-    "dpright",
-    "misc1",
-    "paddle1",
-    "paddle2",
-    "paddle3",
-    "paddle4",
-    "touchpad"
+    "a", NULL,
+    "b", NULL,
+    "x", NULL,
+    "y", NULL,
+    "back", NULL,
+    "guide", NULL,
+    "start", NULL,
+    "leftstick", NULL,
+    "rightstick", NULL,
+    "leftshoulder", NULL,
+    "rightshoulder", NULL,
+    "dpup", NULL,
+    "dpdown", NULL,
+    "dpleft", NULL,
+    "dpright", NULL,
+    "misc1", NULL,
+    /* Keep using paddle1-4 when we generate mapping strings so that they
+     * can be reused with SDL2, but accept rightpaddle1 etc. as input */
```

Comment:
I was mulling this over with the positional button change, and I think we need to keep the mapping strings the same so they can be used across SDL versions. This is especially important for Steam where it might set a mapping in the environment and launch an SDL2 game.

Reply:
Yes, what I was aiming for here is that SDL3 will accept either `paddle1` or `rightpaddle1` in input, but will always generate `paddle1` in output. If we ever reach a point where SDL2 becomes completely obsolete and we expect every SDL2 game to be using sdl2-compat, we can make SDL3 start producing the new names.

Reply:
In each pair of names, the first one is the one we will output, and the second (if not NULL) is an alias that will only be accepted in input.